routing/missioncontrol_store.go

```diff
@@ -304,12 +306,25 @@ func (b *missionControlStore) run() {
 func (b *missionControlStore) storeResults() error {
 	b.queueMx.Lock()
 	l := b.queue
-	b.queue = list.New()
+
+	// Only recreate queue if it's not empty because if it is empty we will
+	// return early from the func without running any updates.
+	isEmpty := l.Len() == 0
+	if !isEmpty {
+		b.queue = list.New()
+	}
 	b.queueMx.Unlock()
 
+	// Return early when no new items will be updated.
+	if isEmpty {
+		return nil
+	}
```

Comment:
Modified. 

Just as clarification, I did it in the original way to avoid adding the two mutex `Unlock()` calls. This is not an issue here because this section has very few lines, but I find that in early returns with mutexes involved, it's easy to make a mistake in the future, by adding an additional clause and forgetting to add the necessary Unlock call.